ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/ConstantStepWiseRateIncreaser.java

```diff
@@ -0,0 +1,150 @@
+package com.netflix.ndbench.plugin.es;
+
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.concurrent.Immutable;
+import java.util.ArrayList;
+
+/**
+ * Ramps a rate from  a specified 'initRate' to a 'finalRate' in constant increments over the course of a
+ * given time period as calls are made to {@link #getRateForGivenClockTime}.
+ * <p>
+ * Internally the class computes a mapping table of clock time values to rates that will be returned by
+ * the aforementioned method.  The mapping table uses a base time of zero to start, but each call to
+ * {@link #getRateForGivenClockTime} will typically be made using the current epoch time (as Long).
+ * Callers of this class should record the epoch time at which they wish to begin the rate increase ramp
+ * and specify this recorded  "baseReferenceTime" as the first argument  to {@link #getRateForGivenClockTime}.
+ * This method will adjust the target instances clock value table by adding the baseReferenceTime to each entry.
+ * <p>
+ * <p>
+ * TODO -  mention how auto-tuning rate increases must not start until the driver begins calling writeSingle, and
+ * how their might be a delay in the initial construction of a ConstantStepWiseRateIncreaser and
+ * the time at which the driver starts making the aforementioned calls.
+ * <p>
+ * TODO -- update this comment.
```

Comment:
Sounds like weird `TODO` :) `TODO` update this comment?

Reply:
updating out of date comments and pushing shortly.